dot/rpc/modules/state.go

```diff
@@ -245,10 +245,24 @@ func (sm *StateModule) GetPairs(_ *http.Request, req *StatePairRequest, res *Sta
 	return nil
 }
 
-// Call isn't implemented properly yet.
-func (sm *StateModule) Call(_ *http.Request, _ *StateCallRequest, _ *StateCallResponse) error {
-	_ = sm.networkAPI
-	_ = sm.storageAPI
+// Call makes a call to the runtime.
+func (sm *StateModule) Call(_ *http.Request, req *StateCallRequest, res *StateCallResponse) error {
+	rt, err := sm.blockAPI.GetRuntime(sm.blockAPI.BestBlockHash())
+	if err != nil {
+		return err
+	}
```

Comment:
The `state_call` RPC method has an optional field `Block`, so if the `StateCallRequest` contains the block hash information we should get a runtime based on it otherwise we use the best block hash 
```suggestion
        blockHash := sm.blockAPI.BestBlockHash()
        if req.Block != nil {
             blockHash = *req.Block
        }
	rt, err := sm.blockAPI.GetRuntime(blockHash)
	if err != nil {
		return err
	}
```

Reply:
Nice catch, thank you!